infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/AsyncStorageModeValidator.java

```diff
@@ -0,0 +1,158 @@
+ * </ul>
+ */
+public class AsyncStorageModeValidator implements WorkspaceAttributeValidator {
+
+  private static final Logger LOG = LoggerFactory.getLogger(AsyncStorageModeValidator.class);
+
+  private final String pvcStrategy;
+  private final boolean allowUserDefinedNamespaces;
+  private final String defaultNamespaceName;
+  private final int runtimesPerUser;
+
+  @Inject
+  public AsyncStorageModeValidator(
+      @Named("che.infra.kubernetes.pvc.strategy") String pvcStrategy,
+      @Named("che.infra.kubernetes.namespace.allow_user_defined")
+          boolean allowUserDefinedNamespaces,
+      @Nullable @Named("che.infra.kubernetes.namespace.default") String defaultNamespaceName,
+      @Named("che.limits.user.workspaces.run.count") int runtimesPerUser) {
+
+    this.pvcStrategy = pvcStrategy;
+    this.allowUserDefinedNamespaces = allowUserDefinedNamespaces;
+    this.defaultNamespaceName = defaultNamespaceName;
+    this.runtimesPerUser = runtimesPerUser;
+  }
+
+  @Override
+  public void validate(Map<String, String> attributes) throws ValidationException {
+    if (parseBoolean(attributes.get(ASYNC_PERSIST_ATTRIBUTE))) {
+      isEphemeralAttributeValidation(attributes);
+      pvcStrategyValidation();
+      alowUserDefinedNamespaceValidation();
+      nameSpaceStrategyValidation();
+      runtimesPerUserValidation();
+    }
+  }
+
+  @Override
+  public void validateUpdate(Map<String, String> existing, Map<String, String> update)
+      throws ValidationException {
+    if (parseBoolean(update.get(ASYNC_PERSIST_ATTRIBUTE))) {
+      if (isEphemeral(existing) || isEphemeral(update)) {
+        pvcStrategyValidation();
+        alowUserDefinedNamespaceValidation();
+        nameSpaceStrategyValidation();
+        runtimesPerUserValidation();
+      } else {
+        String message =
+            "Workspace configuration not valid: Asynchronous storage available only for NOT persistent storage";
+        LOG.warn(message);
+        throw new ValidationException(message);
+      }
+    }
+  }
+
+  private void isEphemeralAttributeValidation(Map<String, String> attributes)
+      throws ValidationException {
+    if (!isEphemeral(attributes)) {
+      String message =
+          "Workspace configuration not valid: Asynchronous storage available only for NOT persistent storage";
+      LOG.warn(message);
+      throw new ValidationException(message);
+    }
+  }
+
+  private void runtimesPerUserValidation() throws ValidationException {
+    if (runtimesPerUser > 1) {
+      String message =
+          format(
+              "Workspace configuration not valid: Asynchronous storage available only if 'che.limits.user.workspaces.run.count' set to 1, but got %s",
+              runtimesPerUser);
+      LOG.warn(message);
+      throw new ValidationException(message);
+    }
+  }
+
+  private void nameSpaceStrategyValidation() throws ValidationException {
+    if (Strings.isNullOrEmpty(defaultNamespaceName)
```

Comment:
This can be evaluated in the constructor just once.

Reply:
https://github.com/eclipse/che/issues/17420